Tools/clinic/clinic.py

```diff
@@ -2111,7 +2116,9 @@ def print_block(
             self,
             block: Block,
             *,
-            core_includes: bool = False
+            core_includes: bool = False,
+            # needed if core_includes is true
+            clinic: Clinic | None = None,
```

Comment:
Should we just pass ``clinic.includes`` here, rather than the full ``clinic`` object?

Reply:
I prefer to pass the Clinic class, it may useful later to get more attributes. But I don't understand well the AC design. IMO the whole idea of passing Clinic here is broken, it should not be the role of a "BlockPrinter" to generate these #include :-( Well, I prefer to move step by step, since AC is complicated and (for me) hard to modify.

Reply:
Hmm, I agree with @AA-Turner here: I'd prefer if we only passed exactly what we need, rather than passing the whole `clinic` object. I think it would make the code more explicit, and make the function easier to understand just by looking at its signature

Reply:
Opened #108581 as a sketch.

A